### PR TITLE
Adds Gmail Toolkit

### PR DESCRIPTION
**Pull request by bborn**
Gmail toolkit for creating drafts, sending emails, searching messages and threads.

### CODE REVIEW

**Comment by agola11** [comment]
Does this work much better than the current zapier toolkit when used with gmail?

**Comment by bborn** [comment]
@agola11  I haven't compared the two that much - but my quick testing with Zapier was that it was pretty slow (but seemed to work ok in terms of doing what you wanted)

File: docs/modules/utils/examples/gmail.ipynb [hunk 1]
```diff
@@ -0,0 +1,90 @@
+{
```
Comment (agola11) [inline docs/modules/utils/examples/gmail.ipynb:1]: This should go in https://github.com/hwchase17/langchain/tree/master/docs/modules/agents/toolkits/examples and follow the same examples there

File: langchain/agents/agent_toolkits/gmail/toolkit.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+)
+from langchain.prompts import PromptTemplate
+
+from datetime import datetime
+
+
+SCOPES = ["https://mail.google.com/"]
+
+
+class GmailToolkit(BaseToolkit):
```
Comment (agola11) [inline langchain/agents/agent_toolkits/gmail/toolkit.py:37]: What we've done in the past is refactor all of the client methods outisde of the toolkit in a separate wrapper (see zapier, sql, etc). Then we've initialized the toolkit with that wrapper. It makes the things a bit more modular. Additionally, it looks like some of these methods can throw. One of the benefits of using an agent for these things is retrying and correcting for errors, so could be worth stringifying the error messages for the tools and allowing the agent to interpret them

**Review by agola11 — COMMENTED** [review]
Just getting back to this, sorry for the delay

**Comment by bborn** [comment]
@agola11 @hwchase17  heads up: I'm going on vacation tomorrow for two weeks, so I won't likely have a chance to make any updates here until I'm back. Feel free to move it along without me if you want, otherwise I'll try to address any comments when I'm back.

**Pull request merged**